### DISCUSSION TITLE
Infinite loop only in preview mode!

### DISCUSSION THREAD

**Discussion by ricardo-fnd** [vercel/community#749] [Help] [answered]
Hello community! Hope you are well.
I have a monorepo with 3 nextjs applications. One of them just serves as a router application, with two `<Link />` to the other 2 apps. This following `next.config.js` its from that router application. (ignore the first rewrite)
<img width="514" alt="Screenshot 2022-07-27 at 13 27 33" src="https://user-images.githubusercontent.com/52207061/181247897-456f36d9-886e-4763-86fd-a47566237efb.png">
Here, as you can see, every `<Link href="/login" />` will rewrite to a new application, maintaining the same URL.
`NEXT_PUBLIC_LOGIN_URL` will have the correct URL in development (http://localhost:3002) and in production (which is the deployment URL on vercel)
This is working fine when I `yarn run dev`, `yarn run start` and even on production!!! BUT it's not working in preview mode, where is when that `VERCEL_URL` is used. 
The reason is not working is because, when I press `<Link href="/login" />` vercel presents a 

> This Request has triggered an infinite loop. 508: INFINITE_LOOP

How is this working in development and production (using the NEXT_PUBLIC_LOGIN_URL) and not in preview mode (using that VERCEL_URL)? Notice that only the `<Link />` is not working, because if I access router application or login application directly, both work fine! 

The only line of code that LOGIN APP has inside `next.config.js` is `basePath:/login`

Thanks guys!

**Comment by amyegan** [comment]
Hi @ricardo-fnd. My best guess, without digging into your repo, is that `NEXT_PUBLIC_LOGIN_URL` is pointing to the **login** app for local and production environments. But your fallback `https://${VERCEL_URL}` is pointing at the **router** app itself which is causing the infinite loop error since it would be rewriting back to itself in that case.

If you need more help investigating, I would be happy to take a look. Just share the repo or a minimal reproduction if you can.

**Reply by ricardo-fnd** [reply]
Yes, I just got that! `VERCEL_URL` is always the **router app**.
I wanted a way to rewrite to the **preview dynamic URL** of login app. Is that possible? Because every new branch will generate a new URL in preview mode.
Thank you @amyegan

**Comment by ricardo-fnd** [comment, marked as answer]
So, as @amyegan said, `VERCEL_URL` was always pointing for the same **router app**. I needed a way to know what was the dynamic URL created by the preview mode.
**VERCEL** creates that URL through the name of the git branch and vercel also provides an environment variable called `VERCEL_GIT_COMMIT_REF` which tells us which branch name created that preview URL, so we just concatenate with the default preview URL.

**Reply by amyegan** [reply]
Thanks for coming back to share this! Awesome solution and I'm glad you found something that works for you 😁
